oauth2client/contrib/django_util/storage.py

```diff
@@ -12,16 +12,68 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from oauth2client.contrib.dictionary_storage import DictionaryStorage
+"""Contains a storage module that stores credentials using the Django
+ORM."""
 
-_CREDENTIALS_KEY = 'google_oauth2_credentials'
+from oauth2client.client import Storage
 
 
-def get_storage(request):
-    # TODO(issue 319): Make this pluggable with different storage providers
-    # https://github.com/google/oauth2client/issues/319
-    """ Gets a Credentials storage object for the Django OAuth2 Helper object
-    :param request: Reference to the current request object
-    :return: A OAuth2Client Storage implementation based on sessions
+class DjangoORMStorage(Storage):
+    """Store and retrieve a single credential to and from the Django datastore.
+
+    This Storage helper presumes the Credentials
+    have been stored as a CredentialsField
+    on a db model class.
     """
-    return DictionaryStorage(request.session, key=_CREDENTIALS_KEY)
+
+    def __init__(self, model_class, key_name, key_value, property_name):
+        """Constructor for Storage.
+
+        Args:
+            model: string, fully qualified name of db.Model model class
+            key_name: string, key name for the entity that has the credentials
+            key_value: string, key value for the entity that has the
+               credentials
+            property_name: string, name of the property that is an
+                           CredentialsProperty
+        """
+        super(DjangoORMStorage, self).__init__()
+        self.model_class = model_class
+        self.key_name = key_name
+        self.key_value = key_value
+        self.property_name = property_name
+
+    def locked_get(self):
+        """Retrieve stored credential.
+
+        Returns:
+            oauth2client.Credentials
+        """
+        credential = None
+
+        query = {self.key_name: self.key_value}
+        entities = self.model_class.objects.filter(**query)
+        if len(entities) > 0:
+            credential = getattr(entities[0], self.property_name)
+            if credential and hasattr(credential, 'set_store'):
```

Comment:
[Never `hasattr`](http://doc.bazaar.canonical.com/developers/code-style.html#hasattr-and-getattr).

Instead, `if getattr(credential, 'set_store', None) is not None:`.